onadata/libs/utils/logger_tools.py

```diff
@@ -275,6 +277,25 @@ def check_submission_permissions(request, xform):
               }))
 
 
+def check_submission_encryption(xform: XForm, xml: bytes) -> NoReturn:
+    """
+    Check that the submission is encrypted or unencrypted depending on the
+    encryption status of an XForm.
+
+    The submission is invalid if the XForm's encryption status is different
+    from the submissions
+    """
+    submission_element = ET.fromstring(xml)
+    encryption_status = submission_element.attrib.get('encrypted') == "yes"
+
+    if encryption_status != xform.encrypted:
+        form_status = "encrypted" if xform.encrypted else "unencrypted"
+        submisison_status = "Encrypted" if encryption_status else "Unencrypted"
+        raise InstanceEncryptionError(
+            f"{submisison_status} submissions are not"
+            f" allowed for {form_status} forms.")
+
```

Comment:
Can we use translation strings here to allow this error message to be translated. And please remember to run the command for updating translation files,